CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ and this project adheres to Rust's notion of
 - `orchard::pczt::Zip32Derivation::extract_account_index`
 
 ### Changed
+- MSRV is now 1.66
```

Comment:
What requires or motivates the MSRV bump in `orchard`? Is it just "parity with `sapling-crypto`"? (And what was the motivation there?)

Reply:
This is the easiest workaround for the `constant_time_eq` MSRV breaking bump; I figure that `1.66` is old enough now that it should basically be fine to do this.